pkg/renderer/sgml/templates.go

```diff
@@ -0,0 +1,196 @@
+	SidebarBlock            string
+	SourceBlock             string
+	SourceBlockContent      string
+	SourceParagraph         string
+	StringElement           string
+	SubscriptText           string
+	SuperscriptText         string
+	Table                   string
+	TocRoot                 string
+	TocSection              string
+	UnorderedList           string
+	VerbatimLine            string
+	VerseBlock              string
+	VerseBlockParagraph     string
+	VerseParagraph          string
+}
+
+type sgmlRenderer struct {
+	functions funcMap
+	templates *Templates
+	prepared  sync.Once
+
+	// Processed templates
+	admonitionBlock         *textTemplate
+	admonitionParagraph     *textTemplate
+	article                 *textTemplate
+	articleHeader           *textTemplate
+	blankLine               *textTemplate
+	blockImage              *textTemplate
+	boldText                *textTemplate
+	calloutList             *textTemplate
+	delimitedBlockParagraph *textTemplate
+	documentDetails         *textTemplate
+	documentAuthorDetails   *textTemplate
+	externalCrossReference  *textTemplate
+	exampleBlock            *textTemplate
+	fencedBlock             *textTemplate
+	footnote                *textTemplate
+	footnoteRef             *textTemplate
+	footnoteRefPlain        *textTemplate
+	footnotes               *textTemplate
+	inlineImage             *textTemplate
+	internalCrossReference  *textTemplate
+	invalidFootnote         *textTemplate
+	italicText              *textTemplate
+	labeledList             *textTemplate
+	labeledListHorizontal   *textTemplate
+	lineBreak               *textTemplate
+	link                    *textTemplate
+	listingBlock            *textTemplate
+	literalBlock            *textTemplate
+	manpageHeader           *textTemplate
+	manpageNameParagraph    *textTemplate
+	monospaceText           *textTemplate
+	orderedList             *textTemplate
+	paragraph               *textTemplate
+	passthroughBlock        *textTemplate
+	preamble                *textTemplate
+	qAndAList               *textTemplate
+	quoteBlock              *textTemplate
+	quoteParagraph          *textTemplate
+	sectionContent          *textTemplate
+	sectionHeader           *textTemplate
+	sectionOne              *textTemplate
+	sidebarBlock            *textTemplate
+	sourceBlock             *textTemplate
+	sourceBlockContent      *textTemplate
+	sourceParagraph         *textTemplate
+	stringElement           *textTemplate
+	subscriptText           *textTemplate
+	superscriptText         *textTemplate
+	table                   *textTemplate
+	tocRoot                 *textTemplate
+	tocSection              *textTemplate
+	unorderedList           *textTemplate
+	verbatimLine            *textTemplate
+	verseBlock              *textTemplate
+	verseBlockParagraph     *textTemplate
+	verseParagraph          *textTemplate
+}
+
+func (sr *sgmlRenderer) prepareTemplates() error {
+	t := sr.templates
+	var err error
+
+	sr.prepared.Do(func() {
```

Comment:
in practice, we should only prepare the templates once. In which case did you need to protect that with a `sync.Once`? In the tests maybe?

Reply:
Render() calls this.  So this ensures that we actually only do this once.  This allows for initialization to be lazy, leading to simpler code elsewhere.

Reply:
ok, fine